### DISCUSSION TITLE
How to render without clearing

### DISCUSSION THREAD

**Discussion by martinrhan** [AvaloniaUI/Avalonia#16003] [Q&A]
I am using a RenderTargetBitmap. I have a piece of code like this
```
using (var context = WorldImage.CreateDrawingContext()) {
    Pen pen = new Pen(Brushes.Black);
    context.DrawLine(pen, previousPoint, currentPoint);
}
```
The purpose of this code is two draw a curve by pairs of points. However, it doesn't work because it clears the image before each draw. How to disable this clearing?

————Update————
I see in https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Media/Imaging/RenderTargetBitmap.cs CreateDrawingContext() can have an argument passed in specifying clear or not. However, in my project IDE told me that method dont have such overload. I am using Avalonia 11.0.10 which is newest.  Why is this happenning?

————Solution————
I solved it by changing the version to 11.1.0-beta2

**Comment by timunie** [comment]
Maybe a WriteableBitmap instead?

**Reply by martinrhan** [reply]
WriteableBitmap does not provide a method like CreateDrawingContext()

**Comment by maxkatz6** [comment]
Reuse drawing context.

**Reply by martinrhan** [reply]
I have to dispose the drawing context after each draw to get the drawing applied. However, I don't think a disposed object can be reused.
